pkg/apis/core/v1/defaults.go

```diff
@@ -166,6 +166,17 @@ func SetDefaults_Service(obj *v1.Service) {
 		// further IPFamilies, IPFamilyPolicy defaulting is in ClusterIP alloc/reserve logic
 		// note: conversion logic handles cases where ClusterIPs is used (but not ClusterIP).
 	}
+
+	if utilfeature.DefaultFeatureGate.Enabled(features.LoadBalancerIPMode) &&
+		obj.Spec.Type == v1.ServiceTypeLoadBalancer {
+
+		for _, ing := range obj.Status.LoadBalancer.Ingress {
+			if ing.IPMode == nil {
+				ipMode := v1.LoadBalancerIPModeVIP
+				ing.IPMode = &ipMode
```

Comment:
Do we want to re-create this variable every time we come across a LoadBalancer Ingress with an `IPMode` value set to nil?

Maybe we can do it once?
```go
ipMode := v1.LoadBalancerIPModeVIP

for _, ing := range obj.Status.LoadBalancer.Ingress {
	if ing.IPMode == nil {
		ing.IPMode = &ipMode
         }
}
```

_or_

Set the value in one line?
```go
for _, ing := range obj.Status.LoadBalancer.Ingress {
	if ing.IPMode == nil {
		ing.IPMode = &v1.LoadBalancerIPModeVIP
         }
}
```

Wdyt?

Reply:
Yeah first one definitely (second one won't work since it's a constant 😄) though in the end I think the compiler is smart enough to optimize it 🤔 but yeah I'll update!